influxql/iterator.go

```diff
@@ -539,3 +585,102 @@ type nilFloatIterator struct{}
 
 func (*nilFloatIterator) Close() error      { return nil }
 func (*nilFloatIterator) Next() *FloatPoint { return nil }
+
+// integerReduceSliceFloatIterator executes a reducer on all points in a window and buffers the result.
+// This iterator receives an integer iterator but produces a float iterator.
+type integerReduceSliceFloatIterator struct {
```

Comment:
Hm, ok. I think if we have both that will be a good compromise. Cast iterators in the case of things like binary expressions, but use this transform iterator in special conditions like `mean()`.